learning/bootstrap.py

```diff
@@ -139,43 +113,29 @@ async def teacher_loop(cfg: DictConfig, mle_log: MLELogger):
 
             # 2- Try to prove each of the conjectures
             examples = []
-            student_results= prove_conjectures(agent_dump, conjectures, theory, premises)
+            student_results = prove_conjectures(agent, conjectures, theory, premises)
```

Comment:
Write into the merge message that we no longer dump the agent, and that thus we don't have the bug any more where the evaluation was done on the agent of the previous iteration.